### PR TITLE
Use host CA to validate proxy cert on tsh login -o id.pam

### LINKED ISSUE #3910: Exporting an identity file prompts the user to accept a host key.
```
$ tsh --proxy=localhost login -o ident.txt
WARNING: You are using insecure connection to SSH proxy https://localhost:3080
Enter password for Teleport user foo:
WARNING: You are using insecure connection to SSH proxy https://proxy.example.com:3080
The authenticity of host 'proxy.example.com:3023' can't be established. Its public key is:
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDRAJUYrhAf55N/zV2GPO3FKrytoF3AKhSQFBK6GsQw1xQBfE25j13RYtdmwDsBXa9wShx2xgAbrI6oom8KCb38YMyTdg14otrRiRliyzPlfuflJmCwqeysDQOuxVwm08KFsXP5vwlcEFM1mPJ/ewScGEZRUGR9MfAp0LTkP0hCjDOKuaR46GK63AiGigAKyV9/dhvGZdPCiagA8ZkPzGdm503IxhDlVD5b2D1GcitgAzbTphTWSy9IPjFzqdWCEn0W8UqAmE5nveeL94ZCi2CVnMckGIE31+vfhKiX5VGVNY0aIbLPWXMNzMbZ/9irDevOmMG8OoEbuPibyl5/vDzl

Are you sure you want to continue (yes/no)? yes

The certificate has been written to ident.txt
```

### PR DESCRIPTION
**Pull request by awly**
When exporting an identity file via tsh login, we won't write the CAs
onto ~/.tsh/known_hosts. But this file is used to validate proxy SSH
cert when dialing it later in the code, after login.

This results in a user getting prompted to verify the proxy public key
manually, even though we have the CA cert in memory.

Manually plumb the trusted SSH CAs to client.HostKeyCallback to get it
validated. Fallback to the existing client.HostKeyCallback if it doesn't
work (which "should never happen").

Fixes https://github.com/gravitational/teleport/issues/3910

### CODE REVIEW

**Comment by awly** [comment]
This is kind of gross, but I couldn't find a way to give an SSH CA to `client.Client` without having it write to disk.
Suggestions welcome!

File: tool/tsh/tsh.go [hunk 1]
```diff
@@ -562,6 +562,43 @@ func setupNoninteractiveClient(tc *client.TeleportClient, key *client.Key) error
...
+							return true
+						}
+					}
+				}
+				return false
+			},
+		}
+		err := checker.CheckHostKey(hostname, remote, hostKey)
+		if err != nil && oldHostKeyCallback != nil {
+			err = oldHostKeyCallback(hostname, remote, hostKey)
```
Comment (russjones) [inline tool/tsh/tsh.go:598]: You can use `trace.NewAggregate` to aggregate and return both errors.

**Review by russjones — APPROVED** [review]

**Comment by fspmarshall** [comment]
> This is kind of gross, but I couldn't find a way to give an SSH CA to `client.Client` without having it write to disk.
> Suggestions welcome!

I think the "elegant" solution to this problem would be to provide an in-memory implementation of the `client.LocalKeyStore` interface.  Looking at how the interface was defined, I think that was always the long-term intention.

**Comment by awly** [comment]
retest this please

**Pull request merged**